encoders/firebase-encoders/src/main/java/com/google/firebase/encoders/json/JsonValueObjectEncoderContext.java

```diff
@@ -192,6 +192,25 @@ JsonValueObjectEncoderContext add(@Nullable Object o) throws IOException, Encodi
       jsonWriter.endArray();
       return this;
     }
+    if (o instanceof Map) {
+      @SuppressWarnings("unchecked")
+      Map<Object, Object> map = (Map<Object, Object>) o;
+      jsonWriter.beginObject();
+      for (Map.Entry<Object, Object> entry : map.entrySet()) {
+        Object key = entry.getKey();
+        try {
+          add((String) key, entry.getValue());
+        } catch (ClassCastException ex) {
+          throw new EncodingException(
+              String.format(
```

Comment:
Rephrase as: "Only String keys are currently supported" to suggest that, while there's a way to encode other types to be map keys (https://developers.google.com/protocol-buffers/docs/proto3#json) it's not supported.